viewing_party/party.py

```diff
@@ -0,0 +1,145 @@
+# WAVE 1
+def create_movie(title, genre, rating):
+    if bool(title) and bool(genre) and bool(rating):
+        new_movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return new_movie
+    else:
+        return None
+
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+        print(user_data) 
+    else:
+        return user_data
+
+
+
+#WAVE 2
+def get_watched_avg_rating(user_data):
+    sum = 0
+    i = 0
+    for rating in user_data["watched"]:
+        if len(user_data["watched"]) > 0: 
+            for i in range(len(user_data["watched"])):
+                rating = user_data["watched"][i]["rating"]
+                sum += rating
+                i += 1
+                average_rating = (sum / len(user_data["watched"]))
+            return average_rating
+    if len(user_data["watched"]) == 0:
+        average_rating = 0.0
+        return average_rating
```

Comment:
It's a little odd to check for the length of the watched list as you're iterating through it. That check should occur before we start to iterate. In addition, there's no need to do an index loop if you're already iterating through each element of a list. When we do a for `element` in `list` and that `element` is a dictionary, we can directly access that dictionary's keys as I've demonstrated below:

```suggestion
    sum = 0
    average_rating = 0.0
    if len(user_data["watched"]) > 0: 
        for movie in user_data["watched"]:
            rating = movie["rating"]
            sum += rating
            average_rating = (sum / len(user_data["watched"]))
    return average_rating
```

Also notice how I have omitted the `else` clause.  In this code, if the length of the watched list is 0 then Python would immediately jump to the return statement and return 0 as the average rating.